core/src/test/scala/integration/kafka/api/ClientIdQuotaTest.scala

```diff
@@ -26,6 +26,10 @@ class ClientIdQuotaTest extends BaseQuotaTest {
   override def producerClientId = "QuotasTestProducer-!@#$%^&*()"
   override def consumerClientId = "QuotasTestConsumer-!@#$%^&*()"
 
+  // The test does not setup a default request limit and thus default to Long.MaxValue
+  // which is the default value used when a quota does not exist.
+  override def defaultRequestQuota: Double = Long.MaxValue
+
```

Comment:
Why do we use Long.MaxValue here and Int.MaxValue in the base class?

Reply:
When the quota limit does not exist, `Long.MaxValue` is returned by default: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ClientQuotaManager.scala#L329. As `ClientIdQuotaTest` does not set the request quota at all, `Long.MaxValue` is the expected value. This is the default quota when not defined.

Whereas, the other tests relying on the base class uses `Int.MaxValue` as a default. It is an arbitrary value here. Any other values which are high enough would work as well.

Reply:
The PR description only mentions the wait for quota change. It would be good to explain the reason for this change there too.

Reply:
I have complemented the description.

Reply:
I don't think `val defaultRequestQuota = Int.MaxValue` in BaseQuotaTest is supposed to be some special value which is different from unlimited quota. I would not be in any case, even if other tests explicitly set request quota to this value.  I think we should change  `val defaultRequestQuota = Int.MaxValue` in BaseQuotaTest to be Long.MaxValue as well. Otherwise, we are making it even more confusing and harder to debug later.

Reply:
if `Int.MaxValue` wasn't used for a specific reason, it is indeed better to use `Long.MaxValue` in all cases.